ludwig/data/preprocessing.py

```diff
@@ -1783,26 +1795,36 @@ def handle_features_with_prompt_config(
 
         dataset_cols[input_col_name] = format_input_with_prompt(
             input_col_name,
-            dataset_cols[input_col_name],
+            dataset_df,
             backend,
             prompt_config["task"],
             retrieval_model=retrieval_model,
             k=k,
             template=prompt_config["template"],
         )
 
+    return dataset_cols
+
 
 def _get_prompt_config(config: ModelConfigDict, input_feature_config: Dict) -> Dict:
-    if "prompt" in config and config["prompt"]["task"] is not None:
-        return config["prompt"]
+    if input_feature_config[TYPE] != TEXT:
+        # Prompt config is only applied to text features
+        return None
 
     preprocessing = input_feature_config["preprocessing"]
-    if "prompt" in preprocessing and preprocessing["prompt"]["task"] is not None:
+    if _has_prompt_section(preprocessing):
```

Comment:
How does this work? Are we accepting `prompt` keys at both the local and global level?

Reply:
The LLM schema requires the prompt at the top-level, and does not allow feature-level prompts.

The ECD schema allows prompts at the feature-level, but not the top level.